bib/cmd/bootc-image-builder/main.go

```diff
@@ -290,6 +290,55 @@ func cmdManifest(cmd *cobra.Command, args []string) error {
 	return nil
 }
 
+func handleAWSFlags(cmd *cobra.Command) (upload bool, err error) {
+	imgTypes, _ := cmd.Flags().GetStringArray("type")
+	region, _ := cmd.Flags().GetString("aws-region")
+	if region == "" {
+		return false, nil
+	}
+	bucketName, _ := cmd.Flags().GetString("aws-bucket")
+
+	if !slices.Contains(imgTypes, "ami") {
+		return false, fmt.Errorf("aws flags set for non-ami image type (type is set to %s)", strings.Join(imgTypes, ","))
+	}
+
+	// check as many permission prerequisites as possible before starting
+	client, err := awscloud.NewDefault(region)
+	if err != nil {
+		return false, err
+	}
+
+	logrus.Info("Checking AWS region access...")
```

Comment:
I just noticed that we only show lgorus.rError so maybe this needs tweaking (info by default sounds reasonable to me)

Reply:
Let's do it in a follow-up?